modules/client-common/src/main/java/org/apache/ignite/client/proto/ClientMessageUnpacker.java

```diff
@@ -413,6 +433,109 @@ public Object unpackObject(int dataType) {
         throw new IgniteException("Unknown client data type: " + dataType);
     }
 
+    /**
+     * Packs an object.
+     *
+     * @return Object array.
+     * @throws IllegalStateException in case of unexpected value type.
+     */
+    public Object[] unpackObjectArray() {
+        assert refCnt > 0 : "Unpacker is closed";
+
+        if (tryUnpackNil())
+            return null;
+
+        int size = unpackInt();
+
+        if (size == 0)
+            return new Object[size];
+
+        Object[] args = new Object[size];
+
+        for (int i = 0; i < size; i++) {
+            if (tryUnpackNil())
+                continue;
+
+            ImmutableValue v = unpackValue();
```

Comment:
We can avoid extra allocations here by switching on `getNextFormat()` and then calling corresponding `unpackX` method.

Reply:
Method was refactored. Now I use getNextFormat().